packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -402,6 +411,99 @@ describeCompat("Summaries", "NoCompat", (getTestObjectProvider) => {
 		await provider.ensureSynchronized();
 		await summaryCollection.waitSummaryAck(container.deltaManager.lastSequenceNumber);
 	});
+
+	async function getNackProm(container: IContainer) {
+		return new Promise<void>((resolve, reject) => {
+			const callback = (_reason, error) => {
+				try {
+					assert.strictEqual(error?.statusCode, 429);
```

Comment:
Opinion: The flow try-assert-catch-reject is kind of confusing, I started to put a comment down calling out that the asserts won't affect the Promise but then saw what's going on.

But I'm guessing your point is to get the `assert` library to generate the `Error` for you.  I was going to suggest just doing `if (...) { reject(...); }` but it's up to you.